src/legacy/core_plugins/visualizations/public/np_ready/public/types/types_service.ts

```diff
@@ -49,18 +53,27 @@ export interface VisType {
 export class TypesService {
   private types: Record<string, VisType> = {};
   private unregisteredHiddenTypes: string[] = [];
+
   public setup() {
-    return {
-      registerVisualization: (registerFn: () => VisType) => {
-        const visDefinition = registerFn();
-        if (this.unregisteredHiddenTypes.includes(visDefinition.name)) {
-          visDefinition.hidden = true;
-        }
+    const registerVisualization = (registerFn: () => VisType) => {
+      const visDefinition = registerFn();
+      if (this.unregisteredHiddenTypes.includes(visDefinition.name)) {
+        visDefinition.hidden = true;
+      }
 
-        if (this.types[visDefinition.name]) {
-          throw new Error('type already exists!');
-        }
-        this.types[visDefinition.name] = visDefinition;
+      if (this.types[visDefinition.name]) {
+        throw new Error('type already exists!');
+      }
+      this.types[visDefinition.name] = visDefinition;
+    };
+    return {
+      createBaseVisualization: (config: any) => {
+        const vis = new BaseVisType(config);
+        registerVisualization(() => vis);
```

Comment:
Why does the interface require a function?

Reply:
its the leftover from VisType registry, needs to be cleaned up in followup